specification/schemaregistry/data-plane/Microsoft.EventHub/preview/2018-01-01-preview/schemaregistry.json

```diff
@@ -1,132 +1,279 @@
 {
   "swagger": "2.0",
   "info": {
-    "version": "2018-01-01-preview",
-    "title": "schemaregistry",
-    "description": "Description of the new service",
-    "x-ms-code-generation-settings": {
-      "name": "schemaregistryClient"
-    }
+    "title": "Azure Schema Registry",
+    "version": "2018-01-01-preview"
   },
-  "host": "management.azure.com",
-  "schemes": ["https"],
-  "consumes": ["application/json"],
-  "produces": ["application/json"],
-  "security": [
-    {
-      "azure_auth": ["user_impersonation"]
-    }
-  ],
-  "securityDefinitions": {
-    "azure_auth": {
-      "type": "oauth2",
-      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
-      "flow": "implicit",
-      "description": "Azure Active Directory OAuth2 Flow",
-      "scopes": {
-        "user_impersonation": "impersonate your user account"
+  "x-ms-parameterized-host": {
+    "hostTemplate": "{endpoint}/$schemagroups",
+    "useSchemePrefix": true,
+    "parameters": [
+      {
+        "$ref": "#/parameters/Endpoint"
       }
-    }
+    ]
   },
+  "schemes": [
+    "https"
+  ],
   "paths": {
-    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.EventHub/operations": {
+    "/getSchemaById/{schema-id}": {
       "get": {
-        "tags": ["Tag1"],
-        "operationId": "OperationGroup_Get",
+        "summary": "Get a registered schema by its unique ID reference.",
+        "description": "Gets a registered schema by its unique ID.  Azure Schema Registry guarantees that ID is unique within a namespace.",
+        "x-ms-examples": {
+          "Get schema by ID": {
+            "$ref": "./examples/OperationSchema_GetById.json"
+          }
+        },
+        "operationId": "Schema_GetById",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "name": "schema-id",
+            "in": "path",
+            "description": "References specific schema in registry namespace.",
+            "required": true,
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "headers": {
+              "Location": {
+                "type": "string",
+                "description": "URL location of schema, identified by schema group, schema name, and version."
+              },
+              "X-Serialization": {
+                "type": "string",
+                "description": "Serialization type for the schema being stored."
+              },
+              "X-Schema-Id": {
+                "type": "string",
+                "description": "References specific schema in registry namespace."
+              },
+              "X-Schema-Id-Location": {
```

Comment:
What is this location header used for as opposed to the `Location` header?

Reply:
@hmlam

Reply:
both X-Schema-Id-Location and Location can uniquely identify a schema resource, just with two different identifier.

- X-Schema-Id-Location uses the schema id and is needed for when you are only given the id (e.g. in Kafka runtime scenarios) - e.g. https://hmlam-bn3-556.servicebus.windows.net:443/$schemagroups/getschemabyid/b95dc5b489fc4a5da5d16d0608dcbecb?api-version=2017-04
- Location by itself uses the schema group + schema name + version as a way to uniquely identify the resource - This is usually more for human readable way to access the resource or for discovery and often easier to use in management scenario. e.g. https://hmlam-bn3-556.servicebus.windows.net:443/$schemagroups/group1/schemas/hmlam1/versions/1?api-version=2017-04

Both header serve the same purposes just uses different information to get the resource.

Reply:
@hmlam So, if `schema id` = `schema group` + `schema name` + `version`, would it make sense to have a request to get either the schema id via 'group + name + version', or get the schema content itself via 'group + name + version'? The closest right now is `getIdByContent` which requires 'group + name + type + content'. Doing some quick maths, it seems like `type` + `content` = `version`. Is that correct?

Reply:
I would say type + content will get you the schema id - that's how runtime would most frequently use because schema id is much lighter weight to embed into runtime data for send/receive.

It is a convenience method to provide a translation between schema id <-> group+schema+version but it is not a necessity in general. once you get once side there is no real need to translate that to the other side to address a resource, and the translation is provided in the headers for that translational reasoning anyway.

I would look at it this way. The header here are just a information convenience to provide additional context about the resource, nothing more - so from a runtime SDK point of view there is no real need to worry about group+schema+version so much.  

- All runtime scenario will likely always uses schema id and never use group+schema+version.
- scenario that uses the group+schema+version will likely have its own discovery mechanics to get the group/schema/version from input/user - e.g. UI or powershell commands have browsing mechanics to form the URL, and doesn't rely on schema id at all.

Reply:
@hmlam Makes sense. Thank you for the additional details! I had a discussion with @arerlend that makes what you saying have solid context now. I'll resolve the conversation.

Reply:
Given fixed schema group + schema name, schema content determines which schema version to return.  Group + name + content -> unique ID.